Automattic/wp-calypso
# Favorite Sites: Add favorite functionality

Related to https://github.com/Automattic/dotcom-forge/issues/4540

## Proposed Changes

This pr adds support for adding a site as a favorite

## Testing Instructions
1. Navigate to the sites dashboard (`/sites`)
2. Follow the instructions mentioned https://github.com/Automattic/jetpack/pull/34171#issue-1997181917
3.  Ensure that you see a new column like below
![2023-11-16T17:40:18,244486368+02:00](https://github.com/Automattic/wp-calypso/assets/497103/9f735f04-bdfd-4bb6-9f0c-75a23339e68c)
4. Ensure that when you toggle the star 2 calls could happen:

- If you add a site as favorite a POST request is happening at `/site-favorite` endpoint
- If you remove a site from your favorites a DELETE request is happening at `/site-favorite` endpoint



## Pre-merge Checklist
